frontend/apps/filemanager/filemanagerbookinfo.lua

```diff
@@ -400,17 +402,26 @@ function BookInfo:setCustomMetadata(file, book_props, prop_key, prop_value)
         os.remove(custom_doc_settings.custom_metadata_file)
         DocSettings:removeSidecarDir(file, util.splitFilePathName(custom_doc_settings.custom_metadata_file))
     else
+        if book_props.pages then
+            local original_props = custom_doc_settings:readSetting("doc_props")
+            if original_props.pages ~= book_props.pages then
+                original_props.pages = book_props.pages
+                custom_doc_settings:saveSetting("doc_props", original_props)
+            end
+        end
         custom_doc_settings:saveSetting("custom_props", custom_props)
         custom_doc_settings:flushCustomMetadata(file)
     end
     book_props.display_title = book_props.display_title or display_title -- restore
     -- in memory
     prop_value = prop_value or custom_doc_settings:readSetting("doc_props")[prop_key] -- set custom or restore original
     book_props[prop_key] = prop_value
-    if self.ui then -- currently opened document
-        self.ui.doc_props[prop_key] = prop_value
+    local ui = self.ui or require("apps/reader/readerui").instance
+    if ui and ui.document and ui.document.file == file then -- currently opened document
+        ui.doc_props[prop_key] = prop_value
         if prop_key == "title" then -- generate if original is empty
-            self.ui.doc_props.display_title = prop_value or filemanagerutil.splitFileNameType(file)
+            book_props.display_title = prop_value or filemanagerutil.splitFileNameType(file)
```

Comment:
(I assume we don't need to call :extendProps() because prop_value is already the customized title?)